src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -37,29 +43,38 @@ const useFetchChannelHook = () => {
       anon_user_info_color_code: channelListInfo?.anonUserInfoColorCode
     };
 
-    channel.targetName = channelListInfo?.channelName;
-    channel.targetImage = channelListInfo?.channelImage;
+    newChannel.targetName = channelListInfo?.channelName;
+    newChannel.targetImage = channelListInfo?.channelImage;
 
-    channel.firstMessage = getFirstMessage(channel?.messages);
+    newChannel.firstMessage = getFirstMessage(channel?.messages);
 
     const isDeletedMessage = channel.firstMessage?.message_type === MESSAGE_TYPE_DELETED;
     if (isDeletedMessage) channel.firstMessage.text = DELETED_MESSAGE_TEXT;
 
-    channel.channel = {...channel};
+    newChannel.channel = {...channel};
     const channelType = channel?.type;
 
     try {
-      const channelList = ChannelList.fromChannelAPI(
-        channel,
-        type[channelType],
-        undefined,
-        anonUserInfo
-      );
-
-      await channelList.saveIfLatest(localDb);
-      refresh('channelList');
+      queue.addJob({
+        label: `saveChannelData-${channel?.id}`,
+        task: () => {
+          return new Promise((resolve) => {
+            const channelList = ChannelList.fromChannelAPI(
+              newChannel,
+              type[channelType],
+              undefined,
+              anonUserInfo
+            );
+
+            channelList.saveIfLatest(localDb).then(() => {
+              refresh('channelList');
+              resolve(true);
+            });
+          });
+        }
+      });
     } catch (e) {
-      console.log('error on helperChannelPromiseBuilder');
+      console.log('error on helperChannelPromiseBuilder', e);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to this section are good for performance and error handling. The use of a queue system (`queue.addJob`) instead of direct database operations improves performance by preventing blocking operations. Also, wrapping the database operation inside a try-catch block is a good practice for error handling.



<!-- This is an auto-generated comment by OSS CodeRabbit -->